lsp-mode.el

```diff
@@ -1530,12 +1530,14 @@ WORKSPACE is the workspace that contains the progress token."
 
        ("report"
         (let ((reporter (lsp-workspace-get-work-done-token token workspace)))
-          (progress-reporter-update reporter (gethash "percentage" value nil))))
+          (when reporter
+            (progress-reporter-update reporter (gethash "percentage" value nil)))))
 
        ("end"
         (let ((reporter (lsp-workspace-get-work-done-token token workspace)))
-          (progress-reporter-done reporter)
-          (lsp-workspace-rem-work-done-token token workspace))))))
+          (when reporter
```

Comment:
FTR better to use when-let in this case.

Reply:
Fix-up in #1603, thanks for the tip